slides/k8s/daemonset.md

```diff
@@ -454,136 +536,101 @@ Of course, option 2 offers more learning opportunities. Right?
 ```wait Please edit the object below```
 ```keys /app: rng```
 ```keys ^J```
-```keys noisactive: "yes"```
+```keys noenabled: "yes"```
 ```keys ^[``` ]
 ```keys :wq```
 ```keys ^J```
 -->
 
 ]
 
----
+This time it should work!
 
-## Checking what we've done
+If we did everything correctly, the web UI shouldn't show any change.
 
-.exercise[
+---
 
-- Check the most recent log line of all `app=rng` pods to confirm that exactly one per node is now active:
-  ```bash
-  kubectl logs -l app=rng --tail 1
-  ```
+## Updating labels
 
-]
+- We want to disable the pod that was created by the deployment
 
-The timestamps should give us a hint about how many pods are currently receiving traffic.
+- All we have to do, is remove the `enabled` label from that pod
 
-.exercise[
+- To identify that pod, we can use its name
 
-- Look at the pods that we have right now:
-  ```bash
-  kubectl get pods
-  ```
+- ... Or rely on the fact that it's the only one with a `pod-template-hash` label
 
-]
+- Good to know:
 
----
+  - `kubectl label ... foo=` doesn't remove a label (it sets it to an empty string)
+
+  - to remove label `foo`, use `kubectl label ... foo-`
 
-## Cleaning up
+  - to change an existing label, we would need to add `--overwrite`
 
-- The pods of the deployment and the "old" daemon set are still running
+---
 
-- We are going to identify them programmatically
+## Removing a pod from the load balancer
 
 .exercise[
 
-- List the pods with `app=rng` but without `isactive=yes`:
+- In one window, check the logs of that pod:
   ```bash
-  kubectl get pods -l app=rng,isactive!=yes
+  POD=$(kubectl get pod -l app=rng,pod-template-hash -o name)
+  kubectl logs --tail 1 --follow $POD
+
   ```
+  (We should see a steady stream of HTTP logs)
 
-- Remove these pods:
+- In another window, remove the label from the pod:
   ```bash
-  kubectl delete pods -l app=rng,isactive!=yes
+  kubectl label pod -l app=rng,pod-template-hash enabled-
```

Comment:
Without standing up a cluster and testing, I'm looking at this and wondering if we actually want a - at the end of the line here: `enabled-` instead of `enabled`.

Reply:
Yes! That's the syntax to _remove_ a label.